akka-javasdk/src/main/scala/akka/javasdk/impl/AnySupport.scala

```diff
@@ -212,6 +214,22 @@ private[akka] object AnySupport {
       }
     }
 
+  // avoids reflection on each message of a given type
+  private val parserCache = new ConcurrentHashMap[Class[_], Parser[AnyRef]]()
+
+  // Note: T must only be a GeneratedMessageV3, not enforceable through typesystem
+  def decodeJavaProtobuf[T](payload: BytesPayload, cls: Class[T]): T = {
+    if (payload.contentType.startsWith(AnySupport.DefaultTypeUrlPrefix)) {
+      val parser =
+        parserCache.computeIfAbsent(cls, cls => cls.getMethod("parser").invoke(null).asInstanceOf[Parser[AnyRef]])
```

Comment:
How does this compare to Akka core's ProtobufSerializer that is looking for method "parseFrom" ?